picard/coverart/__init__.py

```diff
@@ -158,7 +158,7 @@ def next_in_queue(self):
                 finally:
                     if ret != CoverArtProvider.WAIT:
                         self.next_in_queue()
-                    return
+                return
```

Comment:
This `return` was in the `finally` block, now it is outside (well, I didn't audit the whole code to see what it changes or not).

Reply:
In the normal flow this is the same, the return runs after the exception handling. But the former hid the exception completely. It shouldn't matter in this case, as we are catching `BaseException`  anyway. But the difference is visible in this code:

```
>>> def func():
...   try:
...     print("try")
...     raise Exception("foo")
...   finally:
...     print("finally")
...     return
... 
>>> func()
try
finally
```

The raised exception is completely invisible, as if it was catched.

Compare to this:

```
>>> def func2():
...   try:
...     print("try")
...     raise Exception("foo")
...   finally:
...     print("finally")
...   return
... 
>>> func2()
try
finally
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in func2
Exception: foo
```